core/src/com/unciv/models/ruleset/unique/UniqueType.kt

```diff
@@ -367,7 +367,7 @@ enum class UniqueType(
     StatsWhenSpreading("When spreading religion to a city, gain [amount] times the amount of followers of other religions as [stat]", UniqueTarget.Unit, UniqueTarget.Global),
 
     // Attack restrictions
-    CanOnlyAttackUnits("Can only attack [combatantFilter] units", UniqueTarget.Unit),
+    CanOnlyAttackUnits("Can only attack [mapUnitFilter] units", UniqueTarget.Unit),
```

Comment:
Haven't checked, but I'm concerned if this affects any uniques that used this to limit to attacking cities. This seems like the only direct way to spell it out

Reply:
`Cannot attack <vs [all] cities>` ?

Reply:
I see that `cannot attack` currently doesn't respect conditionals, sounds like a precondition

Reply:
> `Cannot attack <vs [all] cities>` ?

No, that's `Cannot attack <vs [non-[all]] cities>` which reads clunky or `Cannot attack <vs [all] units>` which is kinda roundabout

Reply:
I really didn't want to go this route, but we *could* have nested filters.
Just like "non-[]", we can have "[cityFilter] cities" and "[mapUnitFilter] units" be legit combatantFilters
This would mean changing a bunch of actual uniques as well though

Reply:
Side irrelevant question: should empty string be added to Constants.all? Allows for `vs [non-[]] cities`